core/state_processor.go

```diff
@@ -197,8 +197,11 @@ func ProcessBeaconBlockRoot(beaconRoot common.Hash, vmenv *vm.EVM, statedb *stat
 		To:        &params.BeaconRootsAddress,
 		Data:      beaconRoot[:],
 	}
-	vmenv.Reset(NewEVMTxContext(msg), statedb)
-	statedb.AddAddressToAccessList(params.BeaconRootsAddress)
+	txctx := NewEVMTxContext(msg)
+	vmenv.Reset(txctx, statedb)
+	if vmenv.ChainConfig().Rules(vmenv.Context.BlockNumber, true, vmenv.Context.Time).IsEIP2929 {
```

Comment:
Honestly do we really need to add `AddAddressToAccessList` here? System call is supposed to be successful anyway.

It's needed for witness construction, but no reason for gas account.

Just a nitpick